### PR TITLE
[db] Move redis publisher to gitpod-db WEB-621

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

* Depends on https://github.com/gitpod-io/gitpod/pull/18264

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-db-incl6dd4dc0d4d</li>
	<li><b>🔗 URL</b> - <a href="https://mp-db-incl6dd4dc0d4d.preview.gitpod-dev.com/workspaces" target="_blank">mp-db-incl6dd4dc0d4d.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - mp-db-include-publisher-gha.14038</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

**Comment by geropl** [comment]
@easyCZ Can you do the rebase and ping me again once preview is running? :pray:

File: components/server/src/monitoring-endpoints.ts [hunk 1]
```diff
@@ -7,17 +7,18 @@
...
 import * as prometheusClient from "prom-client";
-import { registerDBMetrics } from "@gitpod/gitpod-db/lib";
+import { redisMetricsRegistry, registerDBMetrics } from "@gitpod/gitpod-db/lib";
 import { registerServerMetrics } from "./prometheus-metrics";
 
 @injectable()
 export class MonitoringEndpointsApp {
     public create(): express.Application {
-        const registry = prometheusClient.register;
+        let registry = prometheusClient.register;
```
[the remark this answers is not in the document]
Comment (geropl) [reply components/server/src/monitoring-endpoints.ts:16]: (noticed the same on other recent PRs)

[the remark this answers is not in the document]
Comment (easyCZ) [reply components/server/src/monitoring-endpoints.ts:16]: Forced by our linter

[the remark this answers is not in the document]
Comment (easyCZ) [reply components/server/src/monitoring-endpoints.ts:16]: It's because the merged registry is re-assigned down-below

**Review by geropl — APPROVED** [review]
LGTM, tested and works :heavy_check_mark:

**Pull request merged**